lib/search-suggestions/index.jsx

```diff
@@ -0,0 +1,115 @@
+import TagIcon from '../icons/tag';
+
+export class SearchSuggestions extends Component {
+  static displayName = 'SearchSuggestions';
+
+  static propTypes = {
+    onSearch: PropTypes.func.isRequired,
+    query: PropTypes.string.isRequired,
+    filteredTags: PropTypes.array.isRequired,
+    storeKeyHandler: PropTypes.func.isRequired,
+  };
+
+  state = {
+    selectedItem: 0,
+  };
+
+  componentDidMount() {
+    this.props.storeKeyHandler({
+      next: this.nextItem,
+      prev: this.prevItem,
+      select: this.selectItem,
+    });
+  }
+
+  componentDidUpdate() {
+    this.props.storeKeyHandler({
+      next: this.nextItem,
+      prev: this.prevItem,
+      select: this.selectItem,
+    });
+  }
+
+  nextItem = () => {
+    const { filteredTags } = this.props;
+    const { selectedItem } = this.state;
+    const newItem =
+      selectedItem === filteredTags.length
+        ? filteredTags.length
+        : selectedItem + 1;
+    this.setState({ selectedItem: newItem });
+  };
+
+  prevItem = () => {
+    const { selectedItem } = this.state;
+    const newItem = selectedItem === 0 ? 0 : selectedItem - 1;
+    this.setState({ selectedItem: newItem });
+  };
+
+  selectItem = () => {
+    const { selectedItem } = this.state;
+    const { query, onSearch, filteredTags } = this.props;
+    if (selectedItem === 0) {
+      onSearch(query);
+    } else {
+      onSearch(`tag:${filteredTags[selectedItem - 1].id}`);
+    }
+  };
+
+  render() {
+    const { query, onSearch, filteredTags } = this.props;
+    const { selectedItem } = this.state;
+    const screenReaderLabel = 'Search suggestions';
+    const shouldShowTagSuggestions = query.length > 2;
+
+    return (
+      <div className="search-suggestions" aria-label={screenReaderLabel}>
+        <div
+          id="query"
+          className={
+            selectedItem === 0
+              ? 'search-suggestion-row search-suggestion-row-selected'
+              : 'search-suggestion-row'
+          }
+          onClick={() => onSearch(query)}
+        >
+          <SmallSearchIcon />
+          <div className="search-suggestion">{query}</div>
+        </div>
+        {shouldShowTagSuggestions &&
+          filteredTags.map((tag, index) => (
+            <div
+              key={tag.id}
+              id={tag.id}
+              className={
+                selectedItem === index + 1
+                  ? 'search-suggestion-row search-suggestion-row-selected'
+                  : 'search-suggestion-row'
+              }
+              onClick={() => onSearch(`tag:${tag.id}`)}
+            >
+              <TagIcon />
+              <div className="search-suggestion">
+                {decodeURIComponent(tag.id)}
+              </div>
+            </div>
+          ))}
+      </div>
+    );
+  }
+}
+
+const mapStateToProps = ({ appState: state }, ownProps) => ({
+  filteredTags: state.tags.filter(function(tag) {
+    return tag.id.startsWith(encodeURIComponent(ownProps.query));
```

Comment:
if we have `encodeURIComponent` here in this component it probably means we're neglecting to convert it on entry into the application. did you come across this incidentally or did you anticipate the need for this?

Reply:
I found the need for it in testing... you're saying the query should get encoded upstream somewhere?

Reply:
I think I got it here, is that what you had in mind? Any thoughts on the fact that this input was not formerly encoded? https://github.com/Automattic/simplenote-electron/pull/1648/commits/8a7493ccec8472600b3aa22f7287d19f5557ce70